src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -86,12 +87,20 @@ export const DropdownMenu = <T,>({
     const [isPopupShown, setPopupShown] = useState(false);
     const anchorRef = useRef<HTMLDivElement | null>(null);
 
+    const previousIsPopupShown = usePreviousValue(isPopupShown);
+    useEffect(() => {
```

Comment:
it's breaking change

Reply:
Why?

Reply:
Thus we oblige the user to memoize `onMenuToggle` callback

Reply:
IMHO, onMenuToggle should be called on open/close events, not in useEffect. It would be nice to read this article https://beta.reactjs.org/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers

Reply:
In other words you propose to remember to call `onMenuToggle` each time state is changed?

Reply:
not when state changes, but when toggle event happens).

Reply:
Can you give some example?

Reply:
for example, deleted code in this PR.

Reply:
"Toggle event" happens 6 times, it is very error prone to remember to add onMenuToggle invocatio  each time

Reply:
we can introduce a new callback that combines `setPopupShown` and `onMenuToggle` since they always occur together and use this callback instead of `setPopupShown(value)`

```ts
const togglePopup = useCallback<typeof setPopupShown>((value) => {
    onMenuToggle?.();
    setPopupShown(value);
}, [onMenuToggle]);
```

maybe not a big difference in this particular case, but this one is in fact more straightforward and more readable

Reply:
i have replaced useEffect with custom hook